### PR TITLE
Add intro, auth users, calling sky gear api, error handling, http endpoint, lambda, cron job, static asset JS doc

### PR DESCRIPTION
**Pull request by tszkanlo**
- Introduction to Cloud Functions
- Trigger by Schedule
- Trigger by Client SDK
- Trigger by HTTP Endpoint
- Serving Static Assets
- APIs in Cloud Functions
- Authenticating Users
- Error Handling

Trigger by Database Event still in progress

### CODE REVIEW

**Comment by carmenlau** [comment]
@tszkanlo Found that the travis build is fail with spelling errors, would you mind help fixing? Thanks a lot!!

**Comment by tszkanlo** [comment]
@carmenlau fixed, thank you very much

**Comment by tensiuyan** [comment]
@david90 Could you help take a look at this PR? Z has already tested all the sample codes, and would like to have you give comments on the coding style. Thanks 😄

File: cloud-function/calling-skygear-api/js.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+---
+
+[[toc]]
+
+
+## Skygear Container
+
+Besides using the SDKs to interact with Skygear,
+you can also call the Skygear APIs directly using the Skygear
+`Container` from the cloud code.
```
Comment (david90) [inline cloud-function/calling-skygear-api/js.md]: Should we say `skygearCloud.CloudCodeContainer()` instead?

File: cloud-function/calling-skygear-api/js.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+that would not be possible using the normal API key.
+A typical use case in the cloud code is to impersonate a user for
+creating, altering or querying records.
+
+You can find your master key in your
+[app settings in the Skygear portal][portal-app-settings],
+or obtain your master key through the cloud code by:
+
+```javascript
+masterKey = skygearCloud.settings.masterKey;
```
Comment (david90) [inline cloud-function/calling-skygear-api/js.md:63]: Add example on how to use the master key

File: cloud-function/calling-skygear-api/js.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+Deleting a record can be done in a similar fashion:
+
+```javascript
+const container = new skygearCloud.CloudCodeContainer();
+container.apiKey = 'your-api-key';
+container.endPoint = skygearCloud.settings.skygearEndpoint + '/';
+container.asUserId = 'admin';
+
+container.publicDB.delete({
+	id: 'cdfc7bb4-afd3-464c-a430-c9564c2202cf'
```
Comment (david90) [inline cloud-function/calling-skygear-api/js.md]: Do we need to specify which type of record are we deleting?

File: cloud-function/http-endpoint/python.md [hunk 4]
```diff
@@ -165,32 +165,25 @@ def json_request_handler(request):
 
 To parse data from a JSON request body:
 
-```python
```
Comment (david90) [inline cloud-function/http-endpoint/python.md]: Is this file for Python cloud code?

File: cloud-function/intro-and-deployment/js.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+```
+Do you see `Meow! Thanks!`? If yes it means the deployment is successful.
+You are now good to go. :grinning:
+
+
+## How Cloud Functions Works
+
+Now, let's examine the Cloud Functions example in details.
+
+Functions defined in the Cloud Functions `cloud_code.py` are recognized
```
Comment (david90) [inline cloud-function/intro-and-deployment/js.md]: `index.js`?

**Review by david90 — COMMENTED** [review]
I left some feedback and questions on the PR. @tszkanlo please help take a look if it needs to be addressed. Thanks!

File: cloud-function/calling-skygear-api/js.md (hunk 2 above)
Comment (tszkanlo) [reply cloud-function/calling-skygear-api/js.md:63]: yea, there is one example on demonstrating saving a record on behalf of the authenticated user

**Comment by tszkanlo** [comment]
@david90 i have finished updating the code. sorry that i misadded the python file to git, so i force pushed again this branch.

File: cloud-function/authenticating-users/js.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+calling the function `current_user_id` in the `skygear.utils.context` module.
+
+The current user ID will be useful in database hooks, lambda functions, or
+custom HTTP endpoint (if authenticated). Scheduled tasks are not run with
+any authenticated user so you cannot get an authenticated user from there.
+
+
+```javascript
+skygearCloud.op('create-task', (param) => {
+	const user_id =
```
Comment (david90) [inline cloud-function/authenticating-users/js.md]: @tszkanlo something missing here?

File: cloud-function/lambda/js.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+    @Override
+    public void onLambdaFail(String reason) {
+        // Error Handling
+    }
+});
+```
+
+**iOS SDK**
+
+```obj-c
```
Comment (david90) [inline cloud-function/lambda/js.md:157]: Good if we can also have swift

**Review by david90 — COMMENTED** [review]
@tszkanlo just 2 very small comments, otherwise it looks good to me and ready to be merged.

File: cloud-function/lambda/js.md (hunk 7 above)
Comment (tszkanlo) [reply cloud-function/lambda/js.md:157]: added to both js and python doc

File: cloud-function/authenticating-users/js.md (hunk 6 above)
Comment (tszkanlo) [reply cloud-function/authenticating-users/js.md]: oops, i should delete this section as JS seems do not have method for returning current user id

**Comment by david90** [comment]
Please help add the new pages you've added to the menu bar as well. Thanks @tszkanlo :)

**Comment by david90** [comment]
Since it depend on #116, so let's merge it together at #118.

**Pull request closed without merging**